finesse/gui/temp_control.py

```diff
@@ -0,0 +1,311 @@
+    """Widgets to view the temperature properties."""
+
+    def __init__(self) -> None:
+        """Creates a panel with a graph to monitor the blackbody temperatures."""
+        super().__init__("BB Monitor")
+
+        layout = self._create_controls()
+        self.setLayout(layout)
+
+    def _create_controls(self) -> QGridLayout:
+        """Creates the overall layout for the panel.
+
+        Returns:
+            QGridLayout: The layout containing the figure.
+        """
+        layout = QGridLayout()
+        self._btns = {"hot": QPushButton("Hot BB"), "cold": QPushButton("Cold BB")}
+        self._btns["hot"].clicked.connect(  # type: ignore
+            partial(self._toggle_axis_visibility, name="hot")
+        )
+        self._btns["cold"].clicked.connect(  # type: ignore
+            partial(self._toggle_axis_visibility, name="cold")
+        )
+        self._create_figure()
+
+        layout.addWidget(self._btns["hot"], 0, 0)
+        layout.addWidget(self._btns["cold"], 1, 0)
+        layout.addWidget(self._canvas, 0, 1, 2, 1)
+
+        return layout
+
+    def _create_figure(self) -> None:
+        """Creates the matplotlib figure to be contained within the panel."""
+        self._figure, ax = plt.subplots()
+        self._ax = {"hot": ax}
+        self._canvas = FigureCanvasQTAgg(self._figure)
+
+        # Placeholder callback to test updating of figure
+        self._canvas.mpl_connect("button_press_event", self._update_figure)
+
+        self._figure_num_pts = 10
+        t = [None] * self._figure_num_pts
+        hot_bb_temp = [None] * self._figure_num_pts
+        cold_bb_temp = [None] * self._figure_num_pts
+
+        colours = plt.rcParams["axes.prop_cycle"]
+        hot_colour = colours.by_key()["color"][0]
+        cold_colour = colours.by_key()["color"][1]
+
+        self._ax["hot"].plot(
+            t, hot_bb_temp, color=hot_colour, marker="x", linestyle="-"
+        )
+        self._ax["hot"].set_ylabel("HOT BB", color=hot_colour)
+
+        self._ax["cold"] = self._ax["hot"].twinx()
+        self._ax["cold"].plot(
+            t, cold_bb_temp, color=cold_colour, marker="x", linestyle="-"
+        )
+        self._ax["cold"].set_ylabel("COLD BB", color=cold_colour)
+
+        self._canvas.draw()
+
+    def _toggle_axis_visibility(self, name: str) -> None:
+        """Shows or hides BB plots."""
+        state = self._ax[name].yaxis.get_visible()
+        self._btns[name].setFlat(state)
+        self._ax[name].yaxis.set_visible(not state)
+        self._ax[name].lines[0].set_visible(not state)
+        self._canvas.draw()
+
+    def _update_figure(self, event) -> None:
+        """Updates the matplotlib figure to be contained within the panel."""
+        time = list(self._ax["hot"].lines[0].get_xdata())
+        hot_data = list(self._ax["hot"].lines[0].get_ydata())
+        cold_data = list(self._ax["cold"].lines[0].get_ydata())
+
+        time.pop(0)
+        hot_data.pop(0)
+        cold_data.pop(0)
```

Comment:
This is ok for now, but at some point in the future we might want to be able to view more than the last `n` points (e.g. with a button that lets the user view the whole figure).